docs/tools.md

```diff
@@ -0,0 +1,738 @@
+# Azure MCP Tools Documentation
+
+This document provides a comprehensive reference for all tools available in the Azure MCP Server. The Azure MCP Server provides tools organized by Azure service namespaces, enabling seamless integration between AI agents and Azure services.
+
+## Overview
+
+The Azure MCP Server exposes **130+ tools** across **29+ Azure service namespaces**, providing comprehensive coverage of Azure services from App Configuration to Workbooks. Tools can be accessed in different modes depending on your MCP client's capabilities and requirements.
+
+### Server Modes
+
+- **Namespace Mode (Default)**: Groups tools by service namespace (e.g., "storage", "keyvault", "cosmos")
+- **All Tools Mode**: Exposes each operation as an individual tool
+- **Single Tool Mode**: Single "azure" tool that routes to all operations
+- **Namespace Filtering**: Expose only specific namespaces using `--namespace` flags
+
+## Tool Categories
```

Comment:
Since we're currently adding Namespaces instead of tools, it seems we should group the tools by namespace here in the docs so a user can figure out which tools are in each namespace.